geth/signal/signals.go

```diff
@@ -71,6 +79,8 @@ func Send(signal Envelope) {
 //export NotifyNode
 //nolint: golint
 func NotifyNode(jsonEvent *C.char) {
+	notificationHandlerMutex.RLock()
+	defer notificationHandlerMutex.RUnlock()
```

Comment:
I'd suggest putting `RUnlock` at the end of the function without using `defer` for consistency with other mutex usage introduced in this PR.

Reply:
It's my first commit implementation.
My motivation is if notificationHandler panics notificationHandlerMutex will be unlocked.

Reply:
Ah, that's actually makes perfect sense.